tests/core/test_processor.py

```diff
@@ -1032,3 +1034,57 @@ async def mocked_run(*args: Any, **kwargs: Any) -> List[Event]:
     ]
     for event, expected in zip(tracker.events, expected_events):
         assert event == expected
+
+
+async def test_restart_session(
```

Comment:
optional: A tiny bit clearer, what do you think? 
```suggestion
async def test_restart_triggers_session_start(
```